scipy/sparse/_matrix_io.py

```diff
@@ -111,22 +124,30 @@ def load_npz(file):
             [4, 0, 0]], dtype=int64)
     """
 
-    loaded = np.load(file)
-    try:
-        matrix_format = loaded['format']
-    except KeyError:
-        raise ValueError('The file {} does not contain a sparse matrix.'.format(file))
-
-    try:
-        cls = getattr(scipy.sparse, '{}_matrix'.format(matrix_format))
-    except AttributeError:
-        raise ValueError('Unknown matrix format "{}"'.format(matrix_format))
-
-    if matrix_format in ('csc', 'csr', 'bsr'):
-        return cls((loaded['data'], loaded['indices'], loaded['indptr']), shape=loaded['shape'])
-    elif matrix_format == 'dia':
-        return cls((loaded['data'], loaded['offsets']), shape=loaded['shape'])
-    elif matrix_format == 'coo':
-        return cls((loaded['data'], (loaded['row'], loaded['col'])), shape=loaded['shape'])
-    else:
-        raise NotImplementedError('Load is not implemented for sparse matrix of format {}.'.format(matrix_format))
+    with np.load(file, **PICKLE_KWARGS) as loaded:
+        try:
+            matrix_format = loaded['format']
+        except KeyError:
+            raise ValueError('The file {} does not contain a sparse matrix.'.format(file))
+
+        matrix_format = matrix_format.item()
+
+        if not isinstance(matrix_format, str):
+            # Play safe with Python 2 vs 3 backward compatibility;
+            # files saved with Scipy < 1.0.0 may contain unicode.
+            matrix_format = matrix_format.decode('ascii')
```

Comment:
> Calling unicode.decode is allowed on Python 2.

Right, but it's harmful - AFAICT, for unicode objects, this line means "throw an error if the string isn't ascii"